### DISCUSSION TITLE
Ply drop-offs modelling using PreVABs

### DISCUSSION THREAD

**Discussion by sange000000** [wenbinyugroup/.github#9] [VABS Related Discussions] [answered]
Hello,

I have started using VABS and PreVABs recently. 
Could someone give me a guideline to model ply drops using PreVABS? I would ideally like to model with a single baseline and giving some condition in the component declaration part.

![images](https://github.com/user-attachments/assets/b1f9ab13-89d7-424f-b707-1149bd1a386a)

Thanks :)

**Comment by sange000000** [comment]
adding to the query above. When I add additional layups within a segment only the first layup in the list shows up and the rest are ignored (I'm doing this to add begin and end to get ply drop offs). Adding joint command returns an error 'unable to parse the file'. I have added the lines from my.xml file below


  <component name="skin">
    <segment name="lp">
      <baseline>lp</baseline>
      <layup direction="left">layup2</layup>
	 <layup direction="left">layup1</layup>
    </segment>
    <segment name="hp">
      <baseline>hp</baseline>
      <layup direction="left">layup1</layup>
    </segment>
    <joint style"2"> hp,lp</joint>
  </component>

**Reply by unitasium** [reply]
Can you provide the file or paste the code you are working on?

**Reply by sange000000** [reply]
[materials.txt](https://github.com/user-attachments/files/16672347/materials.txt)
[xml.txt](https://github.com/user-attachments/files/16672348/xml.txt)
 attaching an example where layer begin is used

**Reply by unitasium** [reply, marked as answer]
In this case, the keyword 'segment' should be 'segments'. In addition, the keyword 'layup_side' may cause issues when I tested it. So you can comment it out and by default the layup will be generated on the left side of the base line. If you want the layup to be on the right side, the workaround is to reverse the base line (the order of the point list).
